### ISSUE TITLE
license tag

### ISSUE THREAD

**Issue by rlhamilton** [docbook/docbook#71]
In June 2011 I did some work for including license information in a docbook article.
http://lists.oasis-open.org/archives/docbook/201106/msg00013.html
The proposal is presented here :
http://article.tree.se/tools/docbook/cc.en.html
and result is displayed here :
http://article.tree.se/tools/docbook/cc.html
There were no comments, indicating that the need was limited.
Is there a better way to do this ? 
legalnotice is one way, but for "standard" licenses there could be an easier way.
Maybe the suggested attributes belong to legalnotice instead of a new tag.
Or is "standard" formatting for "standard" licenses out of the scope of docbook ?
I did an attempt to use legalnotice in 2007 but it was not very successful. 
Perhaps a document with the normal creative commons look is just not feasible.
I could try to improve the proposal if there is any interest, and if I get any indication to what needs to be changed.
Thank you
Fredrik

**Comment by tomschr** [comment]
@rlhamilton: the links to article.tree.se don't work anymore. If you can recover it, maybe it can be attached to this issue?

> There were no comments, indicating that the need was limited.

I'm interested in this feature. ;) I think it's a good idea to add an additional `license` tag into `info` to indicate legal matters in a more machine readable way (in contrast to `legalnotice` which is more for humans). However, I can't remember the details.

On the other hand, could the same effect be achieved by using RDF? I think it is already possible to add RDF information into `info`.

**Comment by kosek** [comment]
Yes, it seems that this should be possible to add by using RDFa, see HTML-like example:

````
This document is licensed under the 
<a prefix="cc: http://creativecommons.org/ns#"
   rel="cc:license"
   href="http://creativecommons.org/licenses/by-nc-nd/3.0/"
   >Creative Commons By-NC-ND License</a>.`
````

So before adding new element we should try solution relying on RDFa. If this suitable we should make this approach more visible. There are a lot of metadata in common use, see schema.org.

**Comment by shudson310** [comment]
Is there a way to express the license (RDFa) as part of the <info> block within <legalnotice>?

**Comment by rlhamilton** [comment]
I sent email to the originator of this request (Fredrik Unger), and he sent me an updated URL to replace the ones above: https://article.tree.se/tools/docbook/cc

Be sure to check out the DocBook source referenced on this page. The source shows the markup he is interested in adding.

In short, he suggests adding a license element to copyright, which currently allows only holder and year as child elements. The license element would contains attributes to identify the type, version, and jurisdiction of the license, and it would also allow a link (through the common linking attributes) and text.

**Comment by tomschr** [comment]
> In short, he suggests adding a license element to copyright, [...]

Here is a short example taken from his homepage:

```xml
<copyright>
  <year>2011</year>
  <holder xlink:href="http://article.tree.se/tools/docbook/cc">Fredrik Unger</holder>
  <holder xlink:href="http://article.tree.se/tools/docbook/cc">John Smith</holder>
  <license type="cc by" v="2.5" jurisdiction="se"/>
  <license type="cl" xlink:href="http://article.tree.se/license">Tree.se License</license>
</copyright>
```

**Comment by kosek** [comment]
If we will introduce license it should be allowed in inline contexts as well. You might be speaking about licensing terms in normal text, or inside legalnotice etc.

**Comment by rlhamilton** [comment]
Fredrik sent me a revised link to replace the broken links from above:

https://article.tree.se/tools/docbook/cc

This page describes what he is suggestion and includes a link to a DocBook file that uses this markup.

**Comment by kosek** [comment]
> Is there a way to express the license (RDFa) as part of the info block within legalnotice?

Yes, something like this:

````xml
<book vocab="http://schema.org/" typeof="CreativeWork" xmlns="http://docbook.org/ns/docbook" xmlns:xlink="http://www.w3.org/1999/xlink" version="5.2">
<info>
<title>Sample book</title>
<legalnotice>
<para>Document is available under
<link property="license" xlink:href="http://creativecommons.org/licenses/by/3.0/de/">CC License</link>...
</para>
</legalnotice>
</info>
....
</book>
````

**Comment by kosek** [comment]
To follow up on RDFa usage. There is special vocabulary Creative Commons Rights Expression Language (https://creativecommons.org/ns) which can be used for describing many copyright and license properties. 

If my understanding is correct then expression that particular `<legalnotice>` element contains license can be done as:

```` xml
<legalnotice vocab="http://creativecommons.org/ns#" typeof="License">
  ...
</legalnotice>
````

This vocabulary can describe other things, including jurisdiction. 

So I think one way to proceed is to create more such examples and say that no change in DocBook schema is needed as information can be expressed in RDFa. Problem is that there is no single RDF vocabulary for licenses, so interchange can be little bit more difficult then with "native" DocBook markup.

Also, OP asked for `<license>` element inside `<copyright>`. This use-case can't be solved by RDFa because there are no elements with human readable content to which you can attach RDFa. The basic idea of RDFa (and microformats and microdata) is that you add few structured bits to human readable information. So in order to fulfill original requirement it's still needed to extend content model of `copyright` element with either `license`element or something more generic to which RDFa could be attached.

**Comment by ndw** [comment]
Moved to https://codeberg.org/DocBook/docbook/issues/71
